src/Catalog/DataProvider/LocalizedCatalogGroupOptionDataProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Gally to newer versions in the future.
+ *
+ * @package   Gally
+ * @author    Gally Team <elasticsuite@smile.fr>
+ * @copyright 2022-present Smile
+ * @license   Open Software License v. 3.0 (OSL-3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Gally\Catalog\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use Gally\Catalog\Model\Source\LocalizedCatalogGroupOption;
+use Gally\Catalog\Repository\CatalogRepository;
+
+class LocalizedCatalogGroupOptionDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    public function __construct(
+        private CatalogRepository $catalogRepository
+    ) {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return LocalizedCatalogGroupOption::class === $resourceClass;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getCollection(string $resourceClass, string $operationName = null, array $context = []): array
+    {
+        $groupOptions = [];
+        foreach ($this->catalogRepository->findAll() as $catalog) {
+            $groupOption['code'] = $groupOption['id'] = $catalog->getCode();
+            $groupOption['label'] = $catalog->getName();
+            $options = [];
+            foreach ($catalog->getLocalizedCatalogs() as $localizedCatalog) {
+                $option['code'] = "localized_catalogs/{$localizedCatalog->getId()}";
```

Comment:
I think I can be strange to receive the code is the `code` field of the group but an object uri in the `code` field of the option ?